crates/oxc_isolated_declarations/src/scope.rs

```diff
@@ -291,8 +294,9 @@ impl<'a> Visit<'a> for ScopeTree<'a> {
     ///             ^^^^^^^^^^^^^^^^^^^^
     /// We need to add both `T` and `K` to the scope
     fn visit_ts_mapped_type(&mut self, ty: &TSMappedType<'a>) {
+        // TODO: doesn't have a scope_id!
+        self.enter_scope(ScopeFlags::empty(), &Cell::default());
```

Comment:
What are we going to do about these?

Reply:
@Dunqing

Reply:
Leave it for me. I will fully refactor it later